### PR TITLE
Abl button stopgap

### PR DESCRIPTION
**Pull request by ckline-tryptic**

### CODE REVIEW

File: backend/app/app/service/abl.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    style = meta.xpath(f"//*[local-name()='book'][@slug='{slug}']")[0].attrib["style"]
+
+    # uuid    
+    collection_xml = get_public_git_file("openstax", repo_name, f"/collections/{slug}.collection.xml", version)
+    collection = etree.fromstring(collection_xml)
+    uuid = collection.xpath("//*[local-name()='uuid']")[0].text
+
+    metadata = {
+        "commit_sha": commit_sha,
+        "commit_timestamp": fixed_timetamp,
```
Comment (TylerZeroMaster) [inline backend/app/app/service/abl.py:49]: This property name should be `committed_at` (that's what the frontend expects).

File: backend/app/app/service/abl.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+from email import header
+import json
+import requests
+import asyncio
+from lxml import etree
+from base64 import b64decode
+
+from app.core import config
+
+headers = {"authorization" : config.GITHUB_API_TOKEN}
```
Comment (TylerZeroMaster) [inline backend/app/app/service/abl.py:10]:
```python
headers = {"authorization" : f'token {config.GITHUB_API_TOKEN}'}
```

[link](https://docs.github.com/en/developers/apps/building-github-apps/identifying-and-authorizing-users-for-github-apps#3-your-github-app-accesses-the-api-with-the-users-access-token)

**Review by m1yag1 — APPROVED** [review]

**Comment by ckline-tryptic** [comment]
## Given:
* A valid osbooks repo
## When:
* Running a successful git preview job in CORGI
## Then:
* The `Copy new ABL entry` button is enabled

## Given:
* A valid osbooks repo
## When:
* Running a successful git preview job in CORGI
* Clicking on the `Copy new ABL entry` button
## Then:
* New abl entry is copied to clipboard
* `approved-books-list.json` is opened in edit mode on github in a new tab

## Given:
* A valid osbooks repo
## When:
* Running a successful git preview job in CORGI
* Clicking on the `Copy new ABL entry` button
* An ABL entry already exists for the book
## Then:
* `approved-books-list.json` is opened to the line number of that books entry

## Given:
* A valid osbooks repo
## When:
* Running an git preview job in CORGI
* Job is unsuccessful
## Then:
* The `Copy new ABL entry` button is not enabled

## Given:
* A valid osbooks repo
## When:
* Running an git preview job in CORGI
* Job is still in progress
## Then:
* The `Copy new ABL entry` button is not enabled

**Comment by omehes** [comment]
testing: deployed corgi to staging

latest changes to main - OK
when job is aborted or failed - Copy New ... link is disabled - OK
when job successful - Copy New... link appears and is enabled - OK

some issues with the link - does not open - under investigation

**Comment by m1yag1** [comment]
The backend container rolled back to the previous version because of failures during start-up. The requests library could not be found. I believe either @TylerZeroMaster or @ckline-tryptic will follow up with a fix to add requests to the main (non-dev) dependency list in poetry.

**Comment by omehes** [comment]
testing: re-deployed corgi to staging (after the fix)

latest changes to main - OK
when job is aborted or failed - Copy New ... link is disabled - OK
when job successful - Copy New... link appears and is enabled and opens abl - OK
abl opens where the entry already exists - ok
abl opens when there is no entry -

**Comment by omehes** [comment]
testing on corgi prod: 20220622.165524
all above steps tested - ok

**Pull request merged**